libs/plugin-cache/src/lib/plugin-cache.ts

```diff
@@ -0,0 +1,65 @@
+import { HandlerArgs, HttpExtPlugin } from '@http-ext/http-ext';
+import { defer, EMPTY, merge, of } from 'rxjs';
+import { shareReplay, takeUntil, tap } from 'rxjs/operators';
+
+import { _applyMetadata } from './add-cache-metadata';
+import { CacheProvider } from './provider';
+import { MemoryCacheProvider } from './providers/memory-provider';
+
+export interface CachePluginOptions {
+  addCacheMetadata: boolean;
```

Comment:
what about `shouldAddCacheMetadata` or just `shouldAddMetadata` as we're in the cache plugin?